### PR TITLE
Fix removing vendor folders (FAQ 1.44)

### LINKED ISSUE #16418: Have a small and empty version of phpMyAdmin
> well, it would be nice to have a "basics" tarball without as much as possible 3rd party stuff not present at all instead the "rm -rf" orgy in phpMyAdmin.spec to begin with

_Originally posted by @thelounge-zz in https://github.com/phpmyadmin/phpmyadmin/issue_comments/712133312_

### LINKED ISSUE #17661: Broken composer package in 5.2.0
### Describe the bug

Cannot install additional package with zipped file.

I know that composer packages are preinstalled, but adding new package was working just fine with version 5.1.1.

### To Reproduce

```
wget -q -c https://files.phpmyadmin.net/phpMyAdmin/5.2.0/phpMyAdmin-5.2.0-english.tar.gz -O - | tar -xz --strip-components 1
composer install / composer require some-package
```

### Expected behavior

Successful install of package.

### What happens

composer fails with:

```
composer install
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.
Your lock file does not contain a compatible set of packages. Please run composer update.

  Problem 1
    - phpmyadmin/phpmyadmin is present at version 1.0.0+no-version-set and cannot be modified by Composer
    - roave/security-advisories dev-latest conflicts with phpmyadmin/phpmyadmin <5.1.3.
    - roave/security-advisories is locked to version dev-latest and an update of this package was not requested.
```

What I think is happening - composer.json is missing explicit version declaration.

Yeap, explicitly adding `    "version": "5.2.0"` to composer.json worked.

Probably worth rethinking zip strategy?

### PR DESCRIPTION
**Pull request by williamdes**
### Description

Ref: https://github.com/Roave/SecurityAdvisories/issues/76
Fixes: https://github.com/phpmyadmin/phpmyadmin/issues/17661
Ref: #16418
Fixes: #16419

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -24,7 +24,19 @@
...
-        "files": ["vendor/phpmyadmin/motranslator/src/functions.php"]
+        "files": ["vendor/phpmyadmin/motranslator/src/functions.php"],
+        "exclude-from-classmap": [
+            "/test/",
+            "/vendor/williamdes/mariadb-mysql-kbs",
+            "/vendor/tecnickcom/tcpdf/tcpdf_barcodes_*.php",
+            "/vendor/tecnickcom/tcpdf/tcpdf_import.php",
+            "/vendor/tecnickcom/tcpdf/tcpdf_parser.php",
+            "/vendor/tecnickcom/tcpdf/include/tcpdf_filters.php",
+            "/vendor/tecnickcom/tcpdf/include/barcodes",
```
Comment (williamdes) [inline composer.json:35]: just a cleanup, we do not need the classes do be in the class map as we do not use them

File: composer.json [hunk 2]
```diff
@@ -24,7 +24,19 @@
         "psr-4": {
             "PhpMyAdmin\\": "libraries/classes"
         },
-        "files": ["vendor/phpmyadmin/motranslator/src/functions.php"]
+        "files": ["vendor/phpmyadmin/motranslator/src/functions.php"],
+        "exclude-from-classmap": [
+            "/test/",
```
Comment (williamdes) [inline composer.json:29]: The files are present when we do composer commands in the release script so so it dumps class maps for it, that's not problematic but very useless and false.
It also breaks my code I wrote to check all classmap files exist

File: doc/faq.rst [hunk 3]
```diff
@@ -690,12 +690,17 @@ A list of files and corresponding functionality which degrade gracefully when re
...
-* :file:`./examples/`
-* :file:`./sql/` (SQL scripts to configure advanced functionality)
+* :file:`./examples/` (configuration examples)
+* :file:`./sql/` (SQL scripts to configure advanced functionalities)
 * :file:`./js/vendor/openlayers/` (GIS visualization)
+* :file:`./js/src/` (Source files to re-build `./js/dist/`)
+* :file:`./js/config/` (Configuration files to re-build `./js/dist/`)
+* :file:`./vendor/williamdes/mariadb-mysql-kbs/` folder (external links to MariaDB and MySQL documentations)
+* :file:`./vendor/code-lts/u2f-php-server/` folder (U2F second factor authentication)
+* :file:`./vendor/pragmarx/google2fa-qrcode/` folder (2FA second factor authentication)
```
Comment (MauricioFauth) [inline doc/faq.rst:703]: I'm not sure about this. Instead of adding more vendor files to remove, I think we should remove the existing ones.

Comment (williamdes) [reply doc/faq.rst:703]: Hmm, no ?
Why should we remove our vendors that make 2FA possible ?

Or do you mean saying that no vendor is removable ?

Comment (MauricioFauth) [reply doc/faq.rst:703]: Any vendor directory. That's not the proper way to "uninstall" a dependency.

Comment (williamdes) [reply doc/faq.rst:703]: That's true, I will change the documentation about it

File: doc/faq.rst [hunk 4]
```diff
@@ -686,16 +686,19 @@ Some users have requested to be able to reduce the size of the phpMyAdmin instal
...
-* :file:`./sql/` (SQL scripts to configure advanced functionality)
-* :file:`./js/vendor/openlayers/` (GIS visualization)
+* :file:`./examples/` (configuration examples)
+* :file:`./sql/` (SQL scripts to configure advanced functionalities)
+* :file:`./js/src/` (Source files to re-build `./js/dist/`)
+* :file:`./js/config/` (Configuration files to re-build `./js/dist/`)
+* Run `composer remove --update-no-dev tecnickcom/tcpdf` (exporting to PDF)
+* Run `composer remove --update-no-dev williamdes/mariadb-mysql-kbs` (external links to MariaDB and MySQL documentations)
+* Run `composer remove --update-no-dev code-lts/u2f-php-server` (U2F second factor authentication)
+* Run `composer remove --update-no-dev pragmarx/google2fa-qrcode` (2FA second factor authentication)
```
Comment (williamdes) [inline doc/faq.rst:701]: @MauricioFauth this is problematic because it will remove all other vendors we installed manually from the suggest space. But because we did revert the changes to composer.json composer does not know why they are here and removes them

```
phpMyAdmin-5.2.1-dev-all-languages$ composer remove williamdes/mariadb-mysql-kbs --dry-run --update-no-dev --no-interaction
williamdes/mariadb-mysql-kbs is not required in your composer.json and has not been removed
./composer.json has been updated
Running composer update williamdes/mariadb-mysql-kbs
Loading composer repositories with package information
Package "williamdes/mariadb-mysql-kbs" listed for update is not locked.
Updating dependencies
Nothing to modify in lock file
Installing dependencies from lock file
Package operations: 0 installs, 0 updates, 8 removals
  - Removing williamdes/mariadb-mysql-kbs (v1.2.13)
  - Removing tecnickcom/tcpdf (6.5.0)
  - Removing pragmarx/google2fa-qrcode (v2.1.1)
  - Removing pragmarx/google2fa (v8.0.1)
  - Removing paragonie/constant_time_encoding (v2.6.3)
  - Removing dasprid/enum (1.0.3)
  - Removing code-lts/u2f-php-server (v1.2.0)
  - Removing bacon/bacon-qr-code (2.0.7)
```

Comment (MauricioFauth) [reply doc/faq.rst:701]: What about using `--no-install` instead of `--update-no-dev`?

File: composer.json (hunk 1 above)
Comment (MauricioFauth) [reply composer.json:35]: Do we really have to maintain this list of excludes?

File: scripts/create-release.sh [hunk 5]
```diff
@@ -468,15 +509,27 @@ do
     PACKAGES_VERSIONS="$PACKAGES_VERSIONS $PACKAGES:$PKG_VERSION"
 done
 
-echo "Installing composer packages '$PACKAGES_VERSIONS'"
+echo "* Installing composer packages '$PACKAGES_VERSIONS'"
 
-composer require --no-interaction --optimize-autoloader --update-no-dev $PACKAGES_VERSIONS
```
Comment (MauricioFauth) [inline scripts/create-release.sh:473]: Why removing `--optimize-autoloader` was necessary?

File: scripts/create-release.sh [hunk 6]
```diff
@@ -284,7 +284,7 @@ delete_phpunit_sandbox() {
         echo 'No phpunit sandbox to delete'
         exit 1;
     fi
-    rm -rf "${TEMP_PHPUNIT_FOLDER}"
+    rm -r "${TEMP_PHPUNIT_FOLDER}"
```
Comment (MauricioFauth) [inline scripts/create-release.sh:287]: Not related to this PR, but could we use the [PHPUnit's PHAR](https://phpunit.de/getting-started/phpunit-9.html) instead of installing it in a sandbox?

File: doc/faq.rst (hunk 4 above)
Comment (williamdes) [reply doc/faq.rst:701]: Well the `--no-install` will make that the package is not removed and composer will warn about it. Basically it does nothing

File: composer.json (hunk 1 above)
Comment (williamdes) [reply composer.json:35]: Yes, unfortunately. It's not really needed about the tcpdf lib, I can remove the lines. At some point I would like to add the files to the vendor exclude removal list. But for now each time I do this composer fails building the class map because they are hard coded files in tcpdf's composer json and if files are missing it gets un happy

File: composer.json [hunk 7]
```diff
@@ -24,7 +24,19 @@
...
+            "/vendor/williamdes/mariadb-mysql-kbs",
+            "/vendor/tecnickcom/tcpdf/tcpdf_barcodes_*.php",
+            "/vendor/tecnickcom/tcpdf/tcpdf_import.php",
+            "/vendor/tecnickcom/tcpdf/tcpdf_parser.php",
+            "/vendor/tecnickcom/tcpdf/include/tcpdf_filters.php",
+            "/vendor/tecnickcom/tcpdf/include/barcodes",
+            "/vendor/pragmarx/google2fa-qrcode",
+            "/vendor/bacon/bacon-qr-code",
+            "/vendor/code-lts/u2f-php-server"
```
Comment (williamdes) [inline composer.json:38]: Other excludes could be removed if we solve the composer remove library problem

File: scripts/create-release.sh (hunk 6 above)
Comment (williamdes) [reply scripts/create-release.sh:287]: Yeah probably if we can extract the version that we use. It's probably more complicated than the sandbox code. Or maybe not

File: doc/faq.rst (hunk 4 above)
Comment (MauricioFauth) [reply doc/faq.rst:701]: What about adding the optional dependencies that are included in the release to the composer.json's `require` with the release script?

Comment (williamdes) [reply doc/faq.rst:701]: Yes, it's an option, but anyone doing packaging will then be blocked because of that. The libs will be required but they still are not. Maybe we could ask composer team some help ?

File: .github/workflows/other-tools.yml [hunk 8]
```diff
@@ -70,9 +70,6 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-yarn-
 
-      - name: Install modules
-        run: yarn install --non-interactive
```
Comment (MauricioFauth) [inline .github/workflows/other-tools.yml:74]: Shouldn't the previous step that caches the Yarn dependencies be removed as well?

**Review by MauricioFauth — APPROVED** [review]

File: .github/workflows/other-tools.yml (hunk 8 above)
Comment (williamdes) [reply .github/workflows/other-tools.yml:74]: Probably not because the install normally uses the system cache

Comment (MauricioFauth) [reply .github/workflows/other-tools.yml:74]: :100:
